File: Java/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0 || nums.length == 1) {
            return nums.length;
        }
        int repeatCount = 0;
        int temp = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == temp) {
                ++ repeatCount;
            } else {
                temp = nums[i];
                nums[i - repeatCount] = temp;
            }
        }
        return nums.length - repeatCount;
    }
}

File: Kotlin/remove-duplicates-from-sorted-array.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        if (nums.size == 0 || nums.size == 1) return nums.size
        var i = 0
        var j = 1
        var newSize = 1
        while (j < nums.size) {
            if (nums[i] < nums[j]) {
                nums[i + 1] = nums[j]
                i++
                j++
                newSize++
            } else {
                j++
            }
        }
        return newSize
    }
}
